github-event-service/pull-request.ts

```diff
@@ -0,0 +1,112 @@
+import * as firebase from 'firebase-admin';
+
+import {getFirebaseInstance} from './firebase-common';
+
+type DocumentReference = firebase.firestore.DocumentReference;
+
+/** The configuration for the branch manager. */
+export interface BranchManagerConfig {
+  enabled: boolean;
+  branches: Array<{
+    branch: string;
+    label: string;
+  }>;
+};
+
+/** 
+ * Limited implementation of the Github pull_request webhook event from
+ * https://developer.github.com/webhooks/#payloads
+ */
+export interface GithubPullRequestEvent { 
+  action: 'labeled' | 'unlabeled' | 'opened' | 'closed' | 'reopened' | 'synchronize';
+  number: number;
+  organization: {
+    login: string;
+  };
+  pull_request: {
+    labels: Array<{
+      name: string
+    }>;
+    merge_commit_sha: string;
+  };
+  repository: {
+    id: number;
+    name: string;
+  };
+};
+
+
+/** The Firebase app. */
+const firestore = getFirebaseInstance().firestore();
+
+
+/**
+ * Handle webhook events from github of the type, `pull_request`.  Based on the action of 
+ * the event, updates or deletes the Pull Request document in firestore.
+ */
+export async function handlePullRequestEvent(event: GithubPullRequestEvent) { 
+  const pullRequestRef = await getOrCreatePullRequestRef(event);
+  const labels = event.pull_request.labels.map(label => label.name);
+  switch (event.action) {
+    case 'labeled':
+      setLabelsForPullRequestRef(pullRequestRef, labels);
+      break;
+    case 'unlabeled':
```

Comment:
Looks like you could use a fallthrough for some of these cases

Reply:
Personally, I don't love the usage of fallthroughs in switch statements.  Is there a strong case for using it?